### PR TITLE
switch from imu message to acceleration in controller

### LINKED ISSUE #66: switch from imu message to acceleration in controller
The need for acceleration data in the controller (derivative for linear velocity controls) should be provided either by `geometry_msgs/AccelStamped` or `geometry_msgs/AccelWithCovarianceStamped`. I believe one of these is already being published by the robot localization package. It should be enabled by setting a param in the configuration yaml. Let's remove the need for imu message and switch to Accel message, hook it up to the robot localization so that our feedback mechanism is provided by "odom" along with the "accel". 

- either make sure robot localization always outputs the accel in body frame, or add transformation to the callback to make sure it is transformed back to the body frame.

### PR DESCRIPTION
**Pull request by ozanhakantunca**
Closes #66

### CODE REVIEW

File: auv_control/auv_control/config/default.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ model:
     - [0, 0, 0, 0, 0.0, 0]
     - [0, 0, 0, 0, 0, 7.98053]
 
-#kp: [0.0, 0.0, 1.5, 0.0, 0.0, 0.0, 1.0, 1.0, 1.0, 5.0, 0.0, 1.0]
```
Comment (senceryazici) [inline auv_control/auv_control/config/default.yaml:26]: lets revert this. this seems to be unrelated.

File: auv_control/auv_control/include/auv_control/controller_ros.h [hunk 2]
```diff
@@ -153,13 +157,21 @@ class ControllerROS {
     latest_command_time_ = ros::Time::now();
   }
 
-  void imu_callback(const sensor_msgs::Imu::ConstPtr& msg) {
-    d_state_(6) = msg->linear_acceleration.x;
-    d_state_(7) = msg->linear_acceleration.y;
-    d_state_(8) = msg->linear_acceleration.z;
+  // void imu_callback(const sensor_msgs::Imu::ConstPtr& msg) {
```
Comment (senceryazici) [inline auv_control/auv_control/include/auv_control/controller_ros.h:160]: lets remove the commented code

File: auv_control/auv_control/include/auv_control/controller_ros.h [hunk 3]
```diff
@@ -250,8 +262,10 @@ void save_parameters() {
   ros::Subscriber odometry_sub_;
   ros::Subscriber cmd_vel_sub_;
   ros::Subscriber cmd_pose_sub_;
-  ros::Subscriber imu_sub_;
```
Comment (senceryazici) [inline auv_control/auv_control/include/auv_control/controller_ros.h:253]: again commented code, lets remove it

File: auv_control/auv_control/include/auv_control/controller_ros.h [hunk 4]
```diff
@@ -75,8 +76,11 @@ class ControllerROS {
         nh_.subscribe("cmd_vel", 1, &ControllerROS::cmd_vel_callback, this);
     cmd_pose_sub_ =
         nh_.subscribe("cmd_pose", 1, &ControllerROS::cmd_pose_callback, this);
-    imu_sub_ = nh_.subscribe("imu", 1, &ControllerROS::imu_callback, this);
-
+    // imu_sub_ =
```
Comment (senceryazici) [inline auv_control/auv_control/include/auv_control/controller_ros.h:79]: commented code, lets remove it

File: auv_control/auv_control/include/auv_control/controller_ros.h [hunk 5]
```diff
@@ -8,10 +8,11 @@
 #include "auv_controllers/controller_base.h"
 #include "auv_controllers/multidof_pid_controller.h"
 #include "geometry_msgs/Wrench.h"
+#include "geometry_msgs/AccelWithCovarianceStamped.h"
 #include "nav_msgs/Odometry.h"
 #include "pluginlib/class_loader.h"
 #include "ros/ros.h"
-#include "sensor_msgs/Imu.h"
+// #include "sensor_msgs/Imu.h"
```
Comment (senceryazici) [inline auv_control/auv_control/include/auv_control/controller_ros.h:15]: commented, lets remove

**Review by senceryazici — COMMENTED** [review]
Good work, this looks almost ready,  with minor modifications required.

File: auv_bringup/config/ekf2.yaml [hunk 6]
```diff
@@ -27,7 +27,7 @@ odom_frame: odom
 base_link_frame: taluy/base_link
 world_frame: odom
 
-odom0: /taluy/localization/odom_imu
+odom0: /taluy/acceleration
```
[the remark this answers is not in the document]
Comment (ozanhakantunca) [reply auv_bringup/config/ekf2.yaml:30]: rostopic info /taluy/acceleration
Type: geometry_msgs/AccelWithCovarianceStamped

Publishers: 
 * /taluy/auv_localization_node (http://ozan:43837/)

Subscribers: 
 * /taluy/auv_control_node (http://ozan:32973/)

I can see that ekf node publishes the "/taluy/acceleration" when I turned back to /taluy/localization/odom_imu in the ekf2.yaml file.

**Review by senceryazici — APPROVED** [review]
good work
